test/unit/ard/geographic/test_geomorphology.py

```diff
@@ -0,0 +1,306 @@
+                with pytest.raises(AssertionError):
+                    self.geomorphology.set_values(
+                        x_data_in=x_data if idx_case != 0 else x_data[:1, :],
+                        y_data_in=y_data if idx_case != 1 else y_data[:1, :],
+                        z_data_in=z_data if idx_case != 2 else z_data[:1, :],
+                        material_data_in=(
+                            material_data if idx_case != 3 else material_data[:1, :]
+                        ),
+                    )
+
+            # reset to a legitimate setup
+            self.geomorphology.set_values(
+                x_data_in=x_data,
+                y_data_in=y_data,
+                z_data_in=z_data,
+                material_data_in=material_data,
+            )
+
+            # override one of the values to be invalid
+            if idx_case == 0:
+                self.geomorphology.x_data = self.geomorphology.x_data[:1, :]
+            elif idx_case == 1:
+                self.geomorphology.y_data = self.geomorphology.y_data[:1, :]
+            elif idx_case == 2:
+                self.geomorphology.z_data = self.geomorphology.z_data[:1, :]
+            else:
+                self.geomorphology.material_data = self.geomorphology.material_data[
+                    :1, :
+                ]
+
+            # make sure check valid raises an exception
+            with subtests.test(f"check_valid bad override assertion test {idx_case}"):
+                with pytest.raises(AssertionError):
+                    assert self.geomorphology.check_valid()
+
+    def test_set_values(self, subtests):
+
+        # create a mesh and try to upload it
+        y_data, x_data = np.meshgrid([-1.0, 0.0, 1.0], [0.0, 2.0])
+        z_data = np.ones_like(x_data)
+
+        # set up a geomorphology grid data object
+        self.geomorphology = ard.geographic.GeomorphologyGridData()
+        # set the values
+        self.geomorphology.set_values(
+            x_data_in=x_data,
+            y_data_in=y_data,
+            z_data_in=z_data,
+        )
+
+        # make sure the values are set in correctly
+        with subtests.test(f"set_values data equivalence tests"):
+            assert np.allclose(self.geomorphology.x_data, x_data)
+            assert np.allclose(self.geomorphology.y_data, y_data)
+            assert np.allclose(self.geomorphology.z_data, z_data)
+            assert np.allclose(self.geomorphology.get_z_data(), z_data)
+            assert np.all(self.geomorphology.get_shape() == x_data.shape)
+            assert self.geomorphology.material_data.size == 1
+            assert self.geomorphology.material_data == "soil"  # default value
+
+        with subtests.test(f"check_valid final test"):
+            assert self.geomorphology.check_valid()  # check if the data is valid
+
+    def test_set_values_material(self, subtests):
+
+        # create a mesh and try to upload it
+        y_data, x_data = np.meshgrid([-1.0, 0.0, 1.0], [0.0, 2.0])
```

Comment:
x and y are swapped. I get that it works this way, but it is an odd way to write it.